frontend/packages/ceph-storage-plugin/src/components/ocs-install/node-list.tsx

```diff
@@ -266,16 +267,22 @@ const CustomNodeTable: React.FC<CustomNodeTableProps> = ({
   };
 
   const makeLabelNodesRequest = (selectedNodes: NodeKind[]): Promise<NodeKind>[] => {
-    return selectedNodes.map((node: NodeKind) => {
-      const patch = [
-        {
-          op: 'add',
-          path: '/metadata/labels/cluster.ocs.openshift.io~1openshift-storage',
-          value: '',
-        },
-      ];
-      return k8sPatch(NodeModel, node, patch);
+    const promises = [];
+    selectedNodes.map((node: NodeKind) => {
+      const nodeLabel = isLabelPresent(node);
+      if (!nodeLabel) {
+        const patch = [
+          {
+            op: 'add',
+            path: '/metadata/labels/cluster.ocs.openshift.io~1openshift-storage',
+            value: '',
+          },
+        ];
+        return k8sPatch(NodeModel, node, patch);
+      }
     });
+    promises.push(selectedNodes);
```

Comment:
You are creating an array containing an array of promises.` ([][])`
I would do something in the lines of:
```
const promise = k8sPatch(NodeMode,node,patch);
promises.push(promise);
```